presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteCaseExpressionPredicate.java

```diff
@@ -0,0 +1,223 @@
+                }
+            }
+            return null;
+        }
+
+        private boolean isCaseExpression(RowExpression expression)
+        {
+            return expression instanceof SpecialFormExpression && ((SpecialFormExpression) expression).getForm().equals(SWITCH);
+        }
+
+        private RowExpression processCaseExpression(SwitchExpression switchExpression,
+                                                    RowExpression otherExpression,
+                                                    FunctionHandle functionHandle,
+                                                    String displayName,
+                                                    boolean switchExpressionOnRight)
+        {
+            List<RowExpression> andExpressions = new ArrayList<>();
+            List<RowExpression> invertedOperands = new ArrayList<>();
+
+            for (RowExpression whenClause : switchExpression.whenClauses) {
+                RowExpression operand = switchExpression.caseExpression
+                        .map(x -> equalsExpression(switchExpression.caseExpression.get(), ((SpecialFormExpression) whenClause).getArguments().get(0)))
+                        .orElse(((SpecialFormExpression) whenClause).getArguments().get(0));
+                RowExpression result = ((SpecialFormExpression) whenClause).getArguments().get(1);
+                RowExpression resultExpression;
+                if (switchExpressionOnRight) {
+                    resultExpression = callExpression(otherExpression, result, displayName, functionHandle);
+                }
+                else {
+                    resultExpression = callExpression(result, otherExpression, displayName, functionHandle);
+                }
+                andExpressions.add(LogicalRowExpressions.and(resultExpression, operand));
+                invertedOperands.add(logicalRowExpressions.notCallExpression(operand));
+            }
+
+            switchExpression.elseResult.map(x -> LogicalRowExpressions.and(
+                    switchExpressionOnRight ? callExpression(otherExpression, x, displayName, functionHandle) : callExpression(x, otherExpression, displayName, functionHandle),
+                    LogicalRowExpressions.and(invertedOperands)
+            )).ifPresent(andExpressions::add);
+
+            return LogicalRowExpressions.or(andExpressions);
+        }
+
+        private static class SwitchExpression
+        {
+            private final Optional<RowExpression> caseExpression;
+            private final List<RowExpression> whenClauses;
+            private final Optional<RowExpression> elseResult;
+
+            private SwitchExpression(SpecialFormExpression specialForm)
+            {
+                RowExpression caseExpression = specialForm.getArguments().get(0);
+                boolean isCaseExpressionAvailable = !(caseExpression instanceof ConstantExpression &&
+                        ((ConstantExpression) caseExpression).getValue().equals(true));
```

Comment:
Both SearchedCase and SimpleCase expression both have the same format in RowExpresion, if its a SimpleCase Expression, in RowExpresion in place of the case operand we would have ConstantExpression(true). So I am performing the check to avoid doing true = when condition. 
Similar to this: https://github.com/prestodb/presto/blob/123c2ab8487aa962344601895d98af6fec22ddb3/presto-main/src/main/java/com/facebook/presto/sql/gen/SwitchCodeGenerator.java#L83